File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/Margin.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.property;

import com.urbanairship.json.JsonMap;

import androidx.annotation.NonNull;

public class Margin {
    private final int top;
    private final int bottom;
    private final int start;
    private final int end;

    public Margin(int top, int bottom, int start, int end) {
        this.top = top;
        this.bottom = bottom;
        this.start = start;
        this.end = end;
    }

    @NonNull
    public static Margin fromJson(@NonNull JsonMap json) {
        int top = json.opt("top").getInt(0);
        int bottom = json.opt("bottom").getInt(0);
        int start = json.opt("start").getInt(0);
        int end = json.opt("end").getInt(0);

        return new Margin(top, bottom, start, end);
    }

    public int getTop() {
        return top;
    }

    public int getBottom() {
        return bottom;
    }

    public int getStart() {
        return start;
    }

    public int getEnd() {
        return end;
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/Margin.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.property

import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue
import com.urbanairship.json.optionalField

public class Margin public constructor(
    @JvmField public val top: Int,
    @JvmField public val bottom: Int,
    @JvmField public val start: Int,
    @JvmField public val end: Int
) {

    public companion object {

        private const val KEY_TOP = "top"
        private const val KEY_BOTTOM = "bottom"
        private const val KEY_START = "start"
        private const val KEY_END = "end"

        public var NONE: Margin = Margin(0, 0, 0, 0)

        @Throws(JsonException::class)
        public fun fromJson(json: JsonValue): Margin {
            val content = json.requireMap()

            return Margin(
                top = content.opt(KEY_TOP).getInt(0),
                bottom = content.opt(KEY_BOTTOM).getInt(0),
                start = content.opt(KEY_START).getInt(0),
                end = content.opt(KEY_END).getInt(0)
            )
        }
    }
}
